dev/tasks/gandiva/travis.linux.yml

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+os: linux
+dist: trusty
+sudo: required
+language: cpp
+services:
+  - docker
+
+# don't build twice
+if: tag IS blank
+
+env:
+  global:
+    - PLAT=x86_64
+    - TRAVIS_TAG={{ task.tag }}
+
+before_install:
+  - eval "${MATRIX_EVAL}"
+  - ulimit -c unlimited -S
+  - |
+      sudo bash -c "echo -e 'Acquire::Retries 10; Acquire::http::Timeout \"20\";' > /etc/apt/apt.conf.d/99-travis-retry"
+      sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+      sudo apt-get update -qq
+matrix:
+  fast_finish: true
+  include:
+  - compiler: gcc
+    language: cpp
+    os: linux
+    jdk: openjdk8
+    env:
+    - ARROW_TRAVIS_USE_TOOLCHAIN=1
+    - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+    before_script:
+    - git clone -b {{ arrow.branch }} {{ arrow.remote }} arrow
+    - git -C arrow checkout {{ arrow.head }}
+    - mkdir -p dist
+    - arrow/ci/travis_install_linux.sh
+    - arrow/ci/travis_install_clang_tools.sh
+    - export TRAVIS_BUILD_DIR=$TRAVIS_BUILD_DIR/arrow
+    - arrow/ci/travis_install_toolchain.sh
+    - cd arrow/cpp
+    - mkdir build
+    - cd build
+    - export PATH=$TRAVIS_BUILD_DIR/cpp-toolchain/bin:$PATH
+    - export BOOST_ROOT=$TRAVIS_BUILD_DIR/cpp-toolchain
+    - export PROTOBUF_HOME=$TRAVIS_BUILD_DIR/cpp-toolchain
+    - export LD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/cpp-toolchain/lib:$LD_LIBRARY_PATH
+    - cmake .. -DCMAKE_BUILD_TYPE=Release -DARROW_GANDIVA=ON -DARROW_BUILD_UTILITIES=OFF
+    - make -j4
+    - ctest
```

Comment:
IMO We should create the fat jar first, then install and test it - if it's possible.

Reply:
i am not sure i understand.

This task is to create the linux artifacts packaged in the jar, there will be a similar task for mac and another that combines the two. The fat jar would need the individual artifacts for each OS.

Please let me know if you are proposing another approach.

Reply:
@praveenbingo I probably misunderstand the fat jar creation procedure.
Will We create one jar per platform and another fat one?

Reply:
Yup that is what i am planning currently..please advise if i am doing it incorrectly :)

Reply:
reason i did that is, i want to future proof the upload and not pick individual artifacts to upload, which is why i thought it better to upload a working jar per platform and let the combiner pick and choose how to combine.

Reply:
I thought We build a single fat jar. Whatever, four jars seems better to me.

So can We test the platform dependent jar at the end of the script?

Reply:
`build cpp` -> `build platform dependent jar` -> `test the jar`?

Reply:
Great!